docs/evaluate/development-production-features/temporal-nexus.mdx

```diff
@@ -17,32 +17,96 @@ keywords:
   - temporal cloud nexus
 ---
 
+import { RelatedReadContainer, RelatedReadItem } from '@site/src/components/related-read/RelatedRead';
+
 :::tip SUPPORT, STABILITY, and DEPENDENCY INFO
 
-Temporal Nexus is available in [Public Preview](https://docs.temporal.io/evaluate/development-production-features/release-stages#public-preview) for both self-hosted deployments and Temporal Cloud with the [Temporal Go SDK](https://docs.temporal.io/develop/go).
+**Temporal Nexus** is available in [Public Preview](/evaluate/development-production-features/release-stages#public-preview) for both [Temporal Cloud](/cloud/nexus) and [self-hosted deployments](/production-deployment/self-hosted-guide/nexus).
+- [Temporal Go SDK](/develop/go/nexus) support for Nexus is available in [Public Preview](/evaluate/development-production-features/release-stages#public-preview).
+- [Temporal Java SDK](/develop/java/nexus) support for Nexus is available in [Pre-release](/evaluate/development-production-features/release-stages#pre-release).
 
 :::
 
-Temporal Nexus is a feature of the Temporal platform designed to connect durable executions across team, Namespace, region, and cloud boundaries.*
-It promotes a more modular architecture for sharing a subset of your team’s capabilities through well-defined service API contracts for other teams to use, that abstract underlying Temporal primitives, like Workflows, or execute arbitrary code.
+Nexus allows you to reliably connect your Temporal Applications.
```

Comment:
Good start with this and the following line. Reliable connection, improved modularity, security, and fault isolation.

I'm not sure how Nexus boosts troubleshooting. Maybe it helps localize troubleshooting, which is a fall-out of "improved modularity"? Consider dropping "troubleshooting" from the big wins list?

Also is the "your" necessary here? It's restrictive and may be misleading.

Reply:
> Also is the "your" necessary here? It's restrictive and may be misleading.

fixed

> I'm not sure how Nexus boosts troubleshooting.

It keeps Nexus calls in-band, with more debug info (endpoint, service, op name) in the caller's Workflow event history, and provides bi-directional linking so you can walk the execution from the caller to an underlying Temporal primitive in a handler's Workflow, and navigate back up to the caller. So you essentially don't lose debug context vs. alternative solutions like using gRPC or Kafka to connect Temporal Applications, with out-of-band logging, tracing, etc. that is disjoint from the Workflow Event History.

That said, will replace "troubleshooting" with "debugging", which is what we call it in some of the other Nexus docs that covers Execution Debugging.